site/_data/lib/find.js

```diff
@@ -1,51 +1,42 @@
 const path = require('path');
 
-const cacheKey = Symbol('find-cache');
-const alreadyNotifiedKey = Symbol('already-notified');
+const urlCacheKey = Symbol('find-cache-url');
+const pathCacheKey = Symbol('find-cache-path');
+
+const defaultLocale = 'en';
 
 /**
- * Builds and uses a cache to ensure fast lookup of Eleventy pages.
+ * Builds and uses a cache to ensure fast lookup of Eleventy pages by
+ * a specified key.
  *
  * @param {EleventyCollectionItem[]} collection
- * @param {string} urlToFind
+ * @param {unique symbol} cacheKey
+ * @param {string} key
```

Comment:
Please add a description for params, esp. key and needle as they are not self-descriptive

Reply:
Done, also made types more safe.